extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -389,6 +397,34 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     this.qqid_ = null;
   }
 
+  /** @override */
+  layoutCallback() {
+    const superReturnValue = super.layoutCallback();
+    if (!this.refreshManager_) {
+      // Will initiate the refresh lifecycle iff the slot has been enabled to
+      // do so through an appropriate data attribute, or a page-level meta tag.
+      this.refreshManager_ = new RefreshManager(this, {
+        visiblePercentageMin: 50,
+        totalTimeMin: 0,
+        continuousTimeMin: 1,
+      });
+      if (this.refreshManager_.isRefreshable()) {
```

Comment:
should refreshManager constructor just initiate the refresh cycle?  Then you could minimize this to:

this.refreshManager_ = this.refreshManager_ || new RefreshManager(...);